### PR TITLE
Deprecate ds.lazy() since Dataset is lazy already

### PR DESCRIPTION
**Pull request by jianoaix**
## Why are these changes needed?
https://github.com/ray-project/ray/issues/31639

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by c21 — APPROVED** [review]

File: python/ray/data/dataset.py [hunk 1]
```diff
@@ -4131,6 +4131,10 @@ def get_internal_block_refs(self) -> List[ObjectRef[Block]]:
         self._synchronize_progress_bar()
         return blocks
 
+    @Deprecated(
+        message="Dataset is lazy by default, so this conversion call is no longer "
+        "needed and this API will be removed in future release"
```
Comment (zhe-thoughts) [inline python/ray/data/dataset.py:4136]: Nits (grammar): 

- "Datasets is lazy ..."
- "in a future release"

**Review by zhe-thoughts — APPROVED** [review]
Approved for merging into master pending the nits

File: python/ray/data/dataset.py (hunk 1 above)
Comment (jianoaix) [reply python/ray/data/dataset.py:4136]: Done (the `Dataset` here is referring to this class not the library we provide, so incline to keep it as `Dataset`).

**Comment by zhe-thoughts** [comment]
LGTM, feel free to merge

**Pull request merged**